packages/langium/src/grammar/validation/validator.ts

```diff
@@ -877,18 +877,13 @@ export class LangiumGrammarValidator {
     checkOperatorMultiplicitiesForMultiAssignments(rule: ast.ParserRule, accept: ValidationAcceptor): void {
         // for usual parser rules AND for fragments, but not for data type rules!
         if (!rule.dataType) {
-            this.checkOperatorMultiplicitiesForMultiAssignmentsLogic([rule.definition], accept);
+            this.checkOperatorMultiplicitiesForMultiAssignmentsIndependent([rule.definition], new Map(), accept);
         }
     }
 
-    private checkOperatorMultiplicitiesForMultiAssignmentsLogic(startNodes: AstNode[], accept: ValidationAcceptor): void {
-        // new map to store usage information of the assignments
-        const map: Map<string, AssignmentUse> = new Map();
-
-        // top-down traversal for all starting nodes
-        for (const node of startNodes) {
-            this.checkAssignmentNumbersForNode(node, 1, map, accept);
-        }
+    private checkOperatorMultiplicitiesForMultiAssignmentsIndependent(startNodes: AstNode[], map: Map<string, AssignmentUse>, accept: ValidationAcceptor): void {
```

Comment:
Just a minor thing: You could flip the last two arguments and initialize `map` via a default value, like
```
..., accept: ValidationAcceptor, map: Map<string, AssignmentUse> = new Map()): void {
```
and skip the map creation in the call above.

Reply:
done